core/runner/src/main/java/com/cognifide/aet/runner/processing/data/RunIndexWrappers/RunIndexWrapper.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * AET
+ *
+ * Copyright (C) 2013 Cognifide Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.cognifide.aet.runner.processing.data.RunIndexWrappers;
+
+import com.cognifide.aet.communication.api.metadata.Test;
+import com.cognifide.aet.communication.api.metadata.Url;
+import com.cognifide.aet.communication.api.wrappers.MetadataRunDecorator;
+import com.cognifide.aet.communication.api.wrappers.Run;
+import java.util.ArrayList;
+import java.util.Optional;
+
+public abstract class RunIndexWrapper {
+
+  protected Run objectToRunWrapper = null;
+
+  RunIndexWrapper(Run objectToRunWrapper){
+    this.objectToRunWrapper = objectToRunWrapper;
+  }
+
+  public static void cleanUrlFromExecutionData(Url url) {
+    url.setCollectionStats(null);
+    url.getSteps()
+        .forEach(step -> {
+          step.setStepResult(null);
+          if (step.getComparators() != null) {
+            step.getComparators()
+                .forEach(comparator -> {
+                  comparator.setStepResult(null);
+                  comparator.setFilters(new ArrayList<>());
```

Comment:
Should `filters` be cleared here?

Reply:
That's true - it is initial value. However, if this is re-run, doesn't that mean that some information about applied filters is lost?
I mean [`Step`](https://github.com/Cognifide/aet/blob/master/api/communication-api/src/main/java/com/cognifide/aet/communication/api/metadata/Step.java) which is a [part of comparison processing message payload](https://github.com/Cognifide/aet/blob/master/api/communication-api/src/main/java/com/cognifide/aet/communication/api/job/ComparatorJobData.java#L29) consists of [`Comparators`](https://github.com/Cognifide/aet/blob/master/api/communication-api/src/main/java/com/cognifide/aet/communication/api/metadata/Step.java#L48). And each comparator has its list of [`Filters`](https://github.com/Cognifide/aet/blob/master/api/communication-api/src/main/java/com/cognifide/aet/communication/api/metadata/Comparator.java#L33) to apply on the result. Here, this list is cleared.
Is it set later before executing comparison on the rerun processing?